sklearn/neighbors/base.py

```diff
@@ -145,6 +145,8 @@ def _init_params(self, n_neighbors=None, radius=None,
                 alg_check = 'brute'
             else:
                 alg_check = 'ball_tree'
+                if metric not in VALID_METRICS[alg_check]:
+                    alg_check = 'brute'
```

Comment:
This will default to `brute` now if the metric is callable while previously it used to be `ball_tree` right?

Might be better to do 

    elif callable(metric) or metric in VALID_METRICS['ball_tree']:
        ...
    else:
         ....